app.js

```diff
@@ -0,0 +1,116 @@
+const express = require("express");
+const auth = require("./middleware/auth");
+require("express-async-errors");
+
+const app = express();
+
+// .env update
+require("dotenv").config(); // to load the .env file into the process.env object
+const session = require("express-session");
+
+// updating mongoDB
+const MongoDBStore = require("connect-mongodb-session")(session);
+const url = process.env.MONGO_URI;
+
+const store = new MongoDBStore({
+    // may throw an error, which won't be caught
+    uri: url,
+    collection: "mySessions",
+});
+store.on("error", function (error) {
+    console.log(error);
+});
+
+const sessionParms = {
+    secret: process.env.SESSION_SECRET,
+    resave: true,
+    saveUninitialized: true,
+    store: store,
+    cookie: { secure: false, sameSite: "strict" },
+};
+
+if (app.get("env") === "production") {
+    app.set("trust proxy", 1); // trust first proxy
+    sessionParms.cookie.secure = true; // serve secure cookies
+}
+
+app.use(require("body-parser").urlencoded({ extended: true }));
+app.use(session(sessionParms));
+
+// added flash
+app.use(require("connect-flash")());
+
+// Passport initialization
+/* ------------------ 7.19.2024 --------------------- */
+const passport = require("passport");
+const passportInit = require("./passport/passportInit");
+
+passportInit();
+app.use(passport.initialize());
+app.use(passport.session());
+/* -------------------------------------------------- */
+
+app.use(require("./middleware/storeLocals"));
+
+app.set("view engine", "ejs");
+
+// WEEK14 ASSIGNMENT UPDATES /////////////////////////////////////////
+app.get("/", (req, res) => {
+    res.render("index");
+});
+app.use("/sessions", require("./routes/sessionRoutes"));
+/////////////////////////////////////////////////////////////////////
+
+// secret word handling
+const secretWordRouter = require("./routes/secretWord");
+app.use("/secretWord", auth, secretWordRouter);
+
+app.get("/secretWord", (req, res) => {
```

Comment:
With the secretWord router import, shouldn’t need the get and post in app.js anymore. I like the rearranging for clarity though!